File: app/src/main/java/com/buzzybrains/mvvmarchitecture/data/NoteDatabase.java
package com.buzzybrains.mvvmarchitecture.data;

import android.content.Context;
import android.os.AsyncTask;

import com.buzzybrains.mvvmarchitecture.model.Note;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.sqlite.db.SupportSQLiteDatabase;

@Database(entities = {Note.class}, version = 1, exportSchema = false)
public abstract class NoteDatabase extends RoomDatabase {

    private static NoteDatabase instance;
    private static RoomDatabase.Callback roomCallback = new RoomDatabase.Callback() {

        @Override
        public void onCreate(@NonNull SupportSQLiteDatabase db) {
            super.onCreate(db);

            new PopulateDbAsyncTask(instance).execute();
        }
    };

    public static synchronized NoteDatabase getInstance(Context context) {

        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(),
                    NoteDatabase.class, "note_database")
                    .fallbackToDestructiveMigration()
                    .addCallback(roomCallback)
                    .build();
        }

        return instance;

    }

    public abstract NoteDao noteDao();

    private static class PopulateDbAsyncTask extends AsyncTask<Void, Void, Void> {

        private NoteDao noteDao;

        private PopulateDbAsyncTask(NoteDatabase db) {
            noteDao = db.noteDao();

        }

        @Override
        protected Void doInBackground(Void... voids) {

            noteDao.insert(new Note("T1", "D1", 1, false));
            noteDao.insert(new Note("T2", "D2", 2, false));
            noteDao.insert(new Note("T3", "D3", 3, false));
            noteDao.insert(new Note("T4", "D4", 4, false));

            return null;

        }
    }
}

File: app/src/main/java/com/buzzybrains/mvvmarchitecture/data/NoteDatabase.kt
package com.buzzybrains.mvvmarchitecture.data

import android.content.Context
import android.os.AsyncTask

import com.buzzybrains.mvvmarchitecture.model.Note
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.sqlite.db.SupportSQLiteDatabase

@Database(entities = [Note::class], version = 1, exportSchema = false)
abstract class NoteDatabase : RoomDatabase() {

    abstract fun noteDao(): NoteDao



    companion object {

        private var instance: NoteDatabase? = null

        private val roomCallback = object : RoomDatabase.Callback() {

            override fun onCreate(db: SupportSQLiteDatabase) {
                super.onCreate(db)

                PopulateDbAsyncTask(instance!!).execute()
            }
        }


        @Synchronized
        @JvmStatic
        fun getInstance(context: Context): NoteDatabase {

            if (instance == null) {
                instance = Room.databaseBuilder(context.applicationContext,
                        NoteDatabase::class.java, "note_database")
                        .fallbackToDestructiveMigration()
                        .addCallback(roomCallback)
                        .build()
            }

            return instance!!

        }
    }
    class PopulateDbAsyncTask  constructor(db: NoteDatabase) : AsyncTask<Void, Void, Void>() {

        private val noteDao: NoteDao

        init {
            noteDao = db.noteDao()

        }

        override fun doInBackground(vararg voids: Void): Void? {

            noteDao.insert(Note("T1", "D1", 1, false))
            noteDao.insert(Note("T2", "D2", 2, false))
            noteDao.insert(Note("T3", "D3", 3, false))
            noteDao.insert(Note("T4", "D4", 4, false))

            return null

        }
    }



}
